crates/consensus/consensus/src/lib.rs

```diff
@@ -127,11 +126,23 @@ pub trait Consensus: Debug + Send + Sync {
     ) -> Result<(), ConsensusError>;
 }
 
+#[cfg(feature = "std")]
+impl std::error::Error for ConsensusError {
```

Comment:
please move it below the enum and use `Option` directly without the full path

Reply:
of course, done

Reply:
whoops missed one